platform/os_windows.go

```diff
@@ -189,6 +206,170 @@ func SetSdnRemoteArpMacAddress() error {
 	return nil
 }
 
+func HasMellanoxAdapater() bool {
+	adapterName, err := getMellanoxAdapterName()
+	if err != nil {
+		log.Errorf("Error while getting mellanox adapter name: %v", err)
+		return false
+	} else if adapterName == "" {
+		log.Printf("No Mellanox adapter found")
+		return false
+	}
+	log.Printf("Name of Mellanox adapter : %v", adapterName)
+	return true
+}
+
+// Regularly monitors the Mellanox PriorityVLANGTag registry value and sets it to desired value if needed
+func MonitorAndSetMellanoxRegKeyPriorityVLANTag() {
+	for {
+		adapterName, err := getMellanoxAdapterName()
+		if err == nil && adapterName != "" {
```

Comment:
some general guidance: try to keep happy path code the least aligned (good explanation here: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88). this means try to return from functions or scope as early as possible.